File: lua/ht/conf/lsp/servers/sourcekit.lua
local CoreLspServer = require("ht.core.lsp.server")

---@type ht.lsp.ServerOpts
local opts = {}

opts.name = "sourcekit"

opts.mason = false

opts.setup = function(on_attach, capabilities)
  require("lspconfig").sourcekit.setup {
    filetypes = { "swift", "objective-c", "objective-cpp" },
    on_attach = on_attach,
    capabilities = capabilities,
  }
end

return CoreLspServer.new(opts)

File: src/conf/external_tools/lsp_servers/sourcekit.ts
import { LspServer } from "@core/model";

export const server = new LspServer({
  name: "sourcekit",
  exe: false,
  setup: (
    _server: LspServer,
    on_attach: () => void,
    capabilities: LuaTable
  ) => {
    luaRequire("lspconfig").sourcekit.setup({
      // cmd: ["xcrun", "--toolchain", "swift", "sourcekit-lsp"],
      cmd: [
        "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp",
      ],
      filetypes: ["swift", "objective-c", "objective-cpp"],
      on_attach: on_attach,
      capabilities: capabilities,
    });
  },
});
